pkg/kritis/metadata/containeranalysis/containeranalysis.go

```diff
@@ -294,3 +296,74 @@ func (c Client) DeleteOccurrence(ID string) error {
 	glog.Infof("executed deletion of occurrence=%s", ID)
 	return c.client.DeleteOccurrence(c.ctx, req)
 }
+
+// Poll discovery occurrence for an image and wait until container analysis
+// finishes. Throws an error if analysis is not successful or timeouts.
+func WaitForVulnzAnalysis(containerImage string, timeout time.Duration) error {
+	// resourceURL := fmt.Sprintf("https://gcr.io/my-project/my-image")
+	// timeout := time.Duration(5) * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	client, err := ca.NewGrafeasV1Beta1Client(ctx)
+	if err != nil {
+		return fmt.Errorf("NewClient: %v", err)
+	}
+	defer client.Close()
+
+	// ticker is used to poll once per second.
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	// Find the discovery occurrence using a filter string.
+	var discoveryOccurrence *grafeas.Occurrence
+	for discoveryOccurrence == nil {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timeout while retrieving discovery occurrence")
+		case <-ticker.C:
+			req := &grafeas.ListOccurrencesRequest{
+				Parent: fmt.Sprintf("projects/%s", getProjectFromContainerImage(containerImage)),
+				// Vulnerability discovery occurrences are always associated with the
+				// PACKAGE_VULNERABILITY note in the "goog-analysis" GCP project.
+				Filter: fmt.Sprintf(`resourceUrl=%q AND noteProjectId="goog-analysis" AND noteId="PACKAGE_VULNERABILITY"`, util.GetResourceURL(containerImage)),
+			}
+			it := client.ListOccurrences(ctx, req)
+			// Only one occurrence should ever be returned by ListOccurrences
+			// and the given filter.
+			result, err := it.Next()
+			if err == iterator.Done {
+				break
+			}
+			if err != nil {
+				return fmt.Errorf("it.Next: %v", err)
+			}
+			if result.GetDiscovered() != nil {
+				discoveryOccurrence = result
+			}
+		}
+	}
+
+	// Wait for the discovery occurrence to enter a terminal state.
```

Comment:
It seems that you can combine this with the previous loop.  After L342 you check the state, and then good terminal => nil, bad terminal => error, else loop on.